### PR TITLE
Add a q param to network supply

### LINKED ISSUE #4558: Use ranges to track unreleased supply accounts
### Problem

The `hedera.mirror.rest.network.unreleasedSupplyAccounts` started out as a small list of accounts but has since swelled to hundreds that makes it hard to configure and verify.

### Solution

Convert `hedera.mirror.rest.network.unreleasedSupplyAccounts` to a list of ranges:

```yaml
hedera:
  mirror:
    rest:
      network:
        unreleasedSupplyAccounts:
          - from: 0.0.A
            to: 0.0.B
          - from: 0.0.D
            to: 0.0.G

### Alternatives

_No response_

### LINKED ISSUE #4561: Support a simplified response for network supply
### Problem

As a cryptocurrency tracker, I'd like to query specifically for either the total supply or the circulating supply and get a single value in plain text.

### Solution

Enhance the `/api/v1/network/supply` API to accept a new `q` parameter that short circuits the current JSON response flow:
* Add a `q=totalcoins` query parameter that returns a `Content-Type: text/plain` response of the `total_supply` value
* Add a `q=circulating` query parameter that returns a `Content-Type: text/plain` response of the `released_supply ` value

#### Examples:
`GET /api/v1/network/supply?q=totalcoins`
`5000000000000000000`

`GET /api/v1/network/supply?q=circulating`
`2338000663614241146`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Add a `q=circulating|totalcoins` parameter to `/api/v1/network/supply` that returns a `text/plain` response
* Change unreleased supply accounts property to use entity ID ranges

**Related issue(s)**:

Fixes #4558
Fixes #4561

**Notes for reviewer**:
I avoid documenting it in OpenAPI because it is a bit hacky, non-standard and intended only for a couple consumers.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/config/application.yml [hunk 1]
```diff
@@ -40,555 +40,20 @@ hedera:
...
-          - 0.0.746
-          - 0.0.747
-          - 0.0.748
-          - 0.0.749
-          - 0.0.750
-          - 0.0.950
+          - from: 0.0.2
+            to: 0.0.2
+          - from: 0.0.42
+            to: 0.0.42
```
Comment (mgoelswirlds) [inline hedera-mirror-rest/config/application.yml:46]: This is nice.

**Review by mgoelswirlds — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

File: hedera-mirror-rest/config/application.yml (hunk 1 above)
Comment (MarcKriguerAtHedera) [reply hedera-mirror-rest/config/application.yml:46]: I have to agree!

**Review by MarcKriguerAtHedera — APPROVED** [review]

**Pull request merged**